### PR TITLE
(DOCSP-9475): Update Sync Overview page for new product

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
JIRA: https://jira.mongodb.org/browse/DOCSP-9475

Staging: https://docs-mongodbcom-staging.corp.mongodb.com/realm/pegasus/DOCSP-9475/sync/sync-overview.html

### CODE REVIEW

File: source/sync/sync-overview.txt [hunk 1]
```diff
@@ -15,108 +15,146 @@ Sync
...
+arrive at the same state when all changes have been synced.
+This is why Realm sync includes **automatic conflict resolution**.
 
 .. _conflict-resolution:
 
 Resolving Conflicts
 -------------------
 
-Realm Sync's conflict resolution engine is deterministic.
+Realm sync's conflict resolution engine is deterministic.
```
Comment (cbush) [inline source/sync/sync-overview.txt:138]: If we want to "deproductize" Sync® in the docs, maybe we can refer to it like this?
```suggestion
Realm's sync conflict resolution engine is deterministic.
```

File: source/sync/sync-overview.txt [hunk 2]
```diff
@@ -131,42 +169,44 @@ CPU-intensive. Too many writers writing on the same realm
...
-- Realm Sync enables offline-first app development by handling network loss and recovery automatically.
-- Realm Sync has a built-in conflict resolution engine that guarantees strong eventual consistency.
-- To get the best performance with Realm Sync, it is essential to understand the different use case profiles and their implications on scalability.
+- Realm sync enables offline-first app development by handling network
+  loss and recovery automatically.
+- Realm sync has a built-in conflict resolution engine that guarantees
+  strong eventual consistency.
+- To get the best performance with Realm sync, it is essential to
+  understand the different use case profiles and their implications
+  on scalability.
```
Comment (cbush) [inline source/sync/sync-overview.txt:212]: Are you formatting these manually or have you found a VS Code plugin that actually works for this?

File: source/sync/sync-overview.txt [hunk 3]
```diff
@@ -15,108 +15,146 @@ Sync
...
-recovery, and conflict resolution automatically.
+Realm sync enables you to share data between
+:term:`Realm clients <Realm client>` and a
+:term:`synced Atlas cluster <synced cluster>`.
+Realm sync handles data serialization,
+network recovery, and conflict resolution automatically.
 
 .. _the-connectivity-problem:
 
 The Connectivity Problem
```
[the remark this answers is not in the document]
Comment (nathan-contino-mongo) [reply source/sync/sync-overview.txt:26]: I've re-worked this section to de-emphasize the "common but complex" use cases and re-emphasize mobile connectivity. Take another look when you get the chance!

File: source/sync/sync-overview.txt (hunk 2 above)
Comment (nathan-contino-mongo) [reply source/sync/sync-overview.txt:212]: manually, unfortunately, though I do use a plugin to automatically respect these when I newline.

**Pull request merged**